code/game/objects/items/weapons/rpd.dm

```diff
@@ -0,0 +1,292 @@
+var/list/dpipelist = list(
+	list("pipename" = "Straight pipe", "id" = 0, "orientations" = 2, "icon" = "conpipe-s"),
+	list("pipename" = "Bent pipe", "id" = 1, "orientations" = 4, "icon" = "conpipe-c"),
+	list("pipename" = "Junction", "id" = 2, "orientations" = 4, "icon" = "conpipe-j1"),
+	list("pipename" = "Y-junction", "id" = 4, "orientations" = 4, "icon" = "conpipe-y"),
+	list("pipename" = "Trunk", "id" = 5, "orientations" = 4, "icon" = "conpipe-t"),
+	list("pipename" = "Bin", "id" = 6, "orientations" = 1, "icon" = "condisposal"),
+	list("pipename" = "Outlet", "id" = 7, "orientations" = 4, "icon" = "outlet"),
+	list("pipename" = "Chute", "id" = 8, "orientations" = 4, "icon" = "intake"))
+var/list/mainmenu = list(
+	list("category" = "Atmospherics", "mode" = 1, "icon" = "wrench"),
+	list("category" = "Disposals", "mode" = 2, "icon" = "recycle"),
+	list("category" = "Rotate", "mode" = 3, "icon" = "rotate-right"),
+	list("category" = "Flip", "mode" = 4, "icon" = "exchange"),
+	list("category" = "Recycle", "mode" = 5, "icon" = "trash"))
+var/list/pipemenu = list(
+	list("pipecategory" = "Normal", "pipemode" = 1),
+	list("pipecategory" = "Supply", "pipemode" = 2),
+	list("pipecategory" = "Scrubber", "pipemode" = 3),
+	list("pipecategory" = "Devices", "pipemode" = 4),
+	list("pipecategory" = "Heat exchange", "pipemode" = 5))
+
+//NanoUI stuff
+
+/obj/item/weapon/rpd/attack_self(mob/user)
+	ui_interact(user)
+
+/obj/item/weapon/rpd/ui_interact(mob/user, ui_key = "main", var/datum/nanoui/ui = null, var/force_open = 1, var/datum/topic_state/state = inventory_state)
+	ui = nanomanager.try_update_ui(user, src, ui_key, ui, force_open)
+	if(!ui)
+		ui = new(user, src, ui_key, "rpd.tmpl", "[name]", 400, 650, state = state)
+		ui.open()
+		ui.set_auto_update(1)
+
+/obj/item/weapon/rpd/ui_data(mob/user, ui_key = "main", datum/topic_state/state = inventory_state)
+	var/data[0]
+	data["dpipelist"] = dpipelist
+	data["iconrotation"] = iconrotation
+	data["mainmenu"] = mainmenu
+	data["mode"] = mode
+	data["pipelist"] = pipelist
+	data["pipemenu"] = pipemenu
+	data["pipetype"] = pipetype
+	data["whatdpipe"] = whatdpipe
+	data["whatpipe"] = whatpipe
+	return data
+
+/obj/item/weapon/rpd/Topic(href, href_list, nowindow, state)
+	..()
+	if(href_list["iconrotation"])
+		iconrotation = text2num(sanitize(href_list["iconrotation"]))
+	else if(href_list["whatpipe"])
+		whatpipe = text2num(sanitize(href_list["whatpipe"]))
+	else if(href_list["whatdpipe"])
+		whatdpipe = text2num(sanitize(href_list["whatdpipe"]))
+	else if(href_list["pipetype"])
+		pipetype = text2num(sanitize(href_list["pipetype"]))
+	else if(href_list["mode"])
+		mode = text2num(sanitize(href_list["mode"]))
+	else
+		return
+	nanomanager.update_uis(src)
+
+//What the RPD actually does
+
+/obj/item/weapon/rpd/afterattack(atom/target, mob/user, proximity)
+	..()
+	var/turf/T = get_turf(target)
+	if(ismob(target) || istype(target, /obj/structure/window) || !proximity || cooldown)
```

Comment:
I probably need to add a check for shuttle turfs here so someone can't fill the shuttle with 100 disposals bins